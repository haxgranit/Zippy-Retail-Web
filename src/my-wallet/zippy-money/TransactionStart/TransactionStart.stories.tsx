import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import TransactionStart from './TransactionStart';
import MAIN_INFO from '../../../stories/MainInfo';
import { SendMoneyStepsEnum } from '../SendMoneyStepsEnum';
import { TransactionTypeEnum } from '../TransactionTypeEnum';

export default {
  title: 'my-wallet/zippy-money/TransactionStart',
  component: TransactionStart,
} as ComponentMeta<typeof TransactionStart>;

const Template: ComponentStory<typeof TransactionStart> = () => (
  <BrowserRouter>
    <TransactionStart
      accountsList={[]}
      contactList={[]}
      errorMessage=""
      handleTriggerTransaction={undefined}
      isProcessing={false}
      mainInfo={MAIN_INFO}
      selectedAccount={0}
      selectedContact={0}
      setMainInfo={jest.fn()}
      setSelectedAccount={jest.fn()}
      setSelectedContact={jest.fn()}
      step={SendMoneyStepsEnum.TRANSACTION_START}
      transactionType={TransactionTypeEnum.SEND}
      setTunnelType={jest.fn()}
    />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};

export const APIError = Template.bind({});
APIError.args = {};
APIError.parameters = {};
