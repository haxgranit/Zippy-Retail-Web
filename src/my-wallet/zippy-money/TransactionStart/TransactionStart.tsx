import { Button, Form, FormControl } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import ZippyPageContainer from '../../../common/ZippyPageContainer';
import { TransactionInterface } from '../TransactionInterface';
import { TransactionTypeEnum } from '../TransactionTypeEnum';
import { TunnelTypeEnum } from '../TunnelTypeEnum';
import { Contact } from '../../../api';
import { formatContactName } from '../../../Helpers';

export default function TransactionStart({
  contactList,
  // accountsList,
  transactionType,
  handleTriggerTransaction,
  isProcessing,
  // errorMessage,
  mainInfo,
  setMainInfo,
  // selectedAccount,
  // setSelectedAccount,
  selectedContact,
  setSelectedContact,
  tunnelType,
  setTunnelType,
}: TransactionInterface) {
  const navigate = useNavigate();

  return (
    <>
      <ZippyPageContainer title="Personal Banking" subTitle="Made Fun With Zippy!">
        <div className="zippy-btn-group btn-group" role="group" aria-label="Zippy Cash">
          <Button
            className={transactionType === TransactionTypeEnum.SEND ? 'active' : ''}
            onClick={() => navigate(`/my-wallet/zippy-money/${TransactionTypeEnum.SEND}/transaction-start`)}
            disabled={transactionType === TransactionTypeEnum.SEND}
          >
            Send Money
          </Button>
          <Button
            className={transactionType === TransactionTypeEnum.REQUEST ? 'active' : ''}
            onClick={() => navigate(`/my-wallet/zippy-money/${TransactionTypeEnum.REQUEST}/transaction-start`)}
            disabled={transactionType === TransactionTypeEnum.REQUEST}
          >
            Request Money
          </Button>
        </div>
        <div className="account-identifier">
          <Form.Select
            className="send-account-select"
            onChange={(event) => setSelectedContact(Number(event.target.value))}
            value={selectedContact}
          >
            <option>Select</option>
            {contactList?.map((item: Contact) => (
              <option key={item.id} value={item.id}>
                {formatContactName(item.firstName, item.lastName)}
              </option>
            ))}
          </Form.Select>
          {/*
          <FormControl
              placeholder="Enter @ZippyUsername or Email"
            />
            <i className="zippy-cash-icon zc-add" />
          */}
        </div>
        <FormControl
          className="amount"
          value={mainInfo.amount}
          type="number"
          step=".01"
          onChange={(evt) => setMainInfo({ ...mainInfo, amount: Number(evt.target.value) })}
        />
        <div className="select-method">
          <Form.Check
            id="zippy-cash"
            type="radio"
            className="actions"
            label="Zippy.Cash"
            name="actions2"
            onChange={() => setTunnelType(TunnelTypeEnum.ZIPPY_CASH)}
            checked={tunnelType === TunnelTypeEnum.ZIPPY_CASH}
          />
          <Form.Check
            id="interac-e-transfer"
            type="radio"
            className="actions"
            label="InteracÂ® e-Transfer"
            name="actions2"
            onChange={() => setTunnelType(TunnelTypeEnum.INTERAC_E_TRANSFER)}
            checked={tunnelType === TunnelTypeEnum.INTERAC_E_TRANSFER}
          />
        </div>
        <div className="action">
          <Button
            className="zippy-btn"
            disabled={isProcessing}
            onClick={handleTriggerTransaction}
          >
            {isProcessing && <div className="loading spinner-border" role="status" />}
            Zipp It
          </Button>
        </div>
      </ZippyPageContainer>
    </>
  );
}
