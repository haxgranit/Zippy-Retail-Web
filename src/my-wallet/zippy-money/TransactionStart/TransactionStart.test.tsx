import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TransactionStart from './TransactionStart';
import MAIN_INFO from '../../../stories/MainInfo';
import { SendMoneyStepsEnum } from '../SendMoneyStepsEnum';
import { TransactionTypeEnum } from '../TransactionTypeEnum';

const ReactTestRenderer = require('react-test-renderer');

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useParams: () => ({
    stepId: 'details',
  }),
  useNavigate: () => mockedUsedNavigate,
}));

describe('TransactionStart Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <TransactionStart
          accountsList={[]}
          contactList={[]}
          errorMessage=""
          handleTriggerTransaction={undefined}
          isProcessing={false}
          mainInfo={MAIN_INFO}
          selectedAccount={0}
          selectedContact={0}
          setMainInfo={jest.fn()}
          setSelectedAccount={jest.fn()}
          setSelectedContact={jest.fn()}
          step={SendMoneyStepsEnum.TRANSACTION_START}
          transactionType={TransactionTypeEnum.SEND}
          setTunnelType={jest.fn()}
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
