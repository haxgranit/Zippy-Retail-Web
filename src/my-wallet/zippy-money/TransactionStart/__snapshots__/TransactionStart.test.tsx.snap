// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionStart Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <div
          className="title"
        >
          Personal Account
        </div>
        <div
          className="catch"
        >
          Made Fun With Zippy!
        </div>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          aria-label="Zippy Cash"
          className="zippy-btn-group btn-group"
          role="group"
        >
          <button
            className="active btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Send Money
          </button>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Request Money
          </button>
        </div>
        <div
          className="contact-selector"
        >
          <div
            className="contact-identifier"
          >
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="Enter @ZippyUsername or Email"
            />
            <button
              className="btn-select-contact btn-transparent btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <i
                className="zippy-cash-icon zc-add"
              />
            </button>
          </div>
          <div
            className="contact-list-wrap"
          />
        </div>
        <input
          className="amount form-control"
          onChange={[Function]}
          placeholder="0.00"
          step=".01"
          type="number"
          value={1000}
        />
        <div
          className="select-method"
        >
          <div
            className="actions form-check"
          >
            <input
              checked={false}
              className="form-check-input"
              disabled={false}
              id="zippy-cash"
              name="actions2"
              onChange={[Function]}
              type="radio"
            />
            <label
              className="form-check-label"
              htmlFor="zippy-cash"
              title=""
            >
              Zippy.Cash
            </label>
          </div>
          <div
            className="actions form-check"
          >
            <input
              checked={false}
              className="form-check-input"
              disabled={false}
              id="interac-e-transfer"
              name="actions2"
              onChange={[Function]}
              type="radio"
            />
            <label
              className="form-check-label"
              htmlFor="interac-e-transfer"
              title=""
            >
              InteracÂ® e-Transfer
            </label>
          </div>
        </div>
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            type="button"
          >
            Zipp It
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
